lib/Doctrine/MongoDB/Aggregation/Expr.php

```diff
@@ -0,0 +1,890 @@
+     * @return self
+     */
+    public function month($expression)
+    {
+        return $this->operator('$month', $expression);
+    }
+
+    /**
+     * Multiplies numbers together and returns the result.
+     *
+     * The arguments can be any valid expression as long as they resolve to numbers.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/multiply/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @param mixed|self $expression3,... Additional expressions
+     * @return self
+     */
+    public function multiply($expression1, $expression2 /* , $expression3, ... */)
+    {
+        return $this->operator('$multiply', func_get_args());
+    }
+
+    /**
+     * Compares two values and returns:
+     * true when the values are not equivalent.
+     * false when the values are equivalent.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/ne/
+     * @param mixed|self $expression1
+     * @param mixed|self $expression2
+     * @return self
+     */
+    public function ne($expression1, $expression2)
+    {
+        return $this->operator('$ne', array($expression1, $expression2));
+    }
+
+    /**
+     * Evaluates a boolean and returns the opposite boolean value.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/not/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function not($expression)
+    {
+        return $this->operator('$not', $expression);
+    }
+
+    /**
+     * Defines an operator and value on the expression.
+     *
+     * If there is a current field, the operator will be set on it; otherwise,
+     * the operator is set at the top level of the query.
+     *
+     * @param string $operator
+     * @param array|self[]|self $expression
+     * @return self
+     */
+    protected function operator($operator, $expression)
+    {
+        if ($this->currentField) {
+            $this->expr[$this->currentField][$operator] = $this->ensureArray($expression);
+        } else {
+            $this->expr[$operator] = $this->ensureArray($expression);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Returns an array of all values that result from applying an expression to each document in a group of documents that share the same group by key.
+     *
+     * @see http://docs.mongodb.org/manual/reference/operator/aggregation/push/
+     * @param mixed|self $expression
+     * @return self
+     */
+    public function push($expression)
+    {
+        return $this->operator('$push', $expression);
+    }
+
+    /**
+     * Ensure that a current field has been set.
+     *
+     * @throws LogicException if a current field has not been set
+     */
+    protected function requiresCurrentField()
+    {
+        if (!$this->currentField) {
+            throw new LogicException('This method requires you set a current field using field().');
```

Comment:
This was copied from the basic query builder method. Ideally, I think we could have `requiresCurrentField()` take an optional descriptive parameter to include in the exception message. As-is, the user needs to go through the backtrace one level to see the method that actually requires `field()`, since we throw from this vaguely named method.

Perhaps we can change `requiresCurrentField()`'s signature to `requiresCurrentField($method = null)` and then use `($method ?: 'This method')` when building the message? If we change it here, I think we can update the basic query builder as well in this same PR (changing both methods in a single commit).